### PR TITLE
[WIP] Extract orb scripts

### PR DESCRIPTION
**Pull request by iynere**
- give shellcheck orb the ability to extract shell scripts from packed orb source code, so orb source can have its shell scripts linted
- also build out this repo a bit (update config, add issue/PR templates, destructure orb)
- failure here is expected, as it's actually a result of `shellcheck` finding issues in this orb's own shell scripts
- the functionality doesn't _quite_ work here, as the extraction logic involves `grep`-ing for certain keywords to know when a particular command block ends, & since those keywords necessarily appear in this orb's own shell scripts, it borks the functionality
- (which is a fairly isolated edge case but worth solving if possible—otherwise we can just use a sample orb repo to test instead of this orb's own source)

### CODE REVIEW

File: src/commands/extract-inline-scripts.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      name: Extract shell scripts
+      shell: /bin/bash
+      command: |
+        readarray ORBSOURCE < <<parameters.target-filepath>>
+
+        i=0
+        until [ $i -eq ${#ORBSOURCE[@]} ]; do
+          LINE="${ORBSOURCE[$i]}"
+          echo "$LINE"
+          if echo "$LINE" | grep "command: |"; then
```
Comment (marcomorain) [inline src/commands/extract-inline-scripts.yml]: This is not a YAML parser, and won't be able to extract strings from YAML files correctly. You need to use a YAML parser here, and extract any string values located at the `jobs: <job-name>: steps: <step> run: command: <script>` .

From the docs, ShellCheck is also only documented to check `bash` and `sh`, so you will need to filter for those in the `shell` key - otherwise this will run shellcheck over scriptes written in `powershell`, `cmd`, `zsh`, etc.

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+### Orb version
```
Comment (marcomorain) [inline .github/ISSUE_TEMPLATE.md]: There are three changes in this PR - can you please split it into three PRs please?

- one to add the github issue template scaffolding
- one to factor out the commands into seperate files
- one to add the new functionality

File: publish-alpha.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
```
Comment (marcomorain) [inline publish-alpha.sh]: You will want to add `set -euo pipefail` here.

File: src/commands/extract-inline-scripts.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    description: Path including the filename to output the packed orb
+
+  debug:
+    type: boolean
+    default: false
+    description: >
+      Extra output for orb developers
+
+steps:
+  - build-tools/install-ci-tools
```
Comment (marcomorain) [inline src/commands/extract-inline-scripts.yml]: It looks to me like the plan here is to use the shellcheck orb as a linter for other orbs as well as for `config.yml` files?

I think the way you have implemented this is inside out - with this way this is written, the `shellcheck` orb has full knowledge of orbs, the circleci command line tools, etc. The shellcheck orb should just be responsible for running shellcheck - it should be the responsibilty of another orb to pass the required scripts to the `shellcheck` orb for checking IMO.

File: src/commands/extract-inline-scripts.yml (hunk 1 above)
Comment (iynere) [reply src/commands/extract-inline-scripts.yml]: true i suppose it'd be much easier to use something like `yq` lol

File: publish-alpha.sh (hunk 3 above)
Comment (iynere) [reply publish-alpha.sh]: this is just designed to be run locally, it's not part of this (or any orb's) config

File: src/commands/extract-inline-scripts.yml (hunk 4 above)
Comment (iynere) [reply src/commands/extract-inline-scripts.yml]: just orbs, i was thinking @marcomorain

Comment (iynere) [reply src/commands/extract-inline-scripts.yml]: hmmmmm i see what you're saying, but im not sure where that functionality would live then, or what that orb would be... i suppose orb-tools or build-tools could have a command to extract shell scripts from yaml, but i'm not sure what else that command would get used for

**Comment by gmemstr** [comment]
Moving this over to #22
